org.mixedrealitytoolkit.uxcore/Button/Experimental/ExperimentalPressableButtonWithDynamicFrontPlate.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Mixed Reality Toolkit Contributors
+// Licensed under the BSD 3-Clause
+
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace MixedReality.Toolkit.UX
+{
+    /// <summary>
+    /// This mono behaviour is used to enable or disable the front plate of a button when the button enters or exists proximity-hovering.
+    /// </summary>
+    [RequireComponent(typeof(PressableButton))]
+    public class ExperimentalPressableButtonWithDynamicFrontPlate : MonoBehaviour
+    {
+        /// <summary>
+        /// Name of the Frontplate GameObject in the prefab.
+        /// </summary>
+        private const string FrontPlateName = "Frontplate";
+
+        /// <summary>
+        /// Stores the FrontPlate's RawImage component if this is an EmptyButton, ActionButton, or CanvasButtonToggleSwitch.  Null otherwise.
+        /// Populated during runtime on this MonoBehaviour Start method.
+        /// </summary>
+        private RawImage frontPlateRawImage = null;
+
+        #region Private Members
+
+        /// <summary>
+        /// Gets the front plate's raw image component if this is an EmptyButton (Experimental), ActionButton (Experimental), or CanvasButtonToggleSwitch (Experimental).  Null otherwise.
+        /// </summary>
+        /// <returns>Reference to this button FrontPlate's RawImage Component.  Null if it doesn't exist or if this is not an EmptyButton (Experimental), ActionButton (Experimental), or CanvasButtonToggleSwitch (Experimental)</returns>
+        internal RawImage GetFrontPlateRawImage()
+        {
+            if (gameObject.tag.Equals("ExperimentalDynamicFrontplate")) //This is a temporary conditional for the experimental dynamic frontplate feature, it will be removed if the community accepts the experimental feature to be included in official release and be integrated as part of PressableButton script
```

Comment:
I see your comment...however even in experimental components we should avoid poor practices.

Please add a serializable field to this class.

Reply:
Mmmmmmmmhhhh poor practice? ... I disagree to call my approach a poor-practice because it empowers users to immediately benefit from this change and removes human-error from the equation.  On the other hand, I believe the approach of using a SerializedField to manually set the Frontplate's RawImage is human-error prone and a waste of time for the user of Buttons with Dynamic Frontplates because he/she has to do something that could be done automatically as in my approach.  For example:  To update CanvasUITearsheedDynamicFrontplate I (or any other user) will have to go through all 90-something buttons in the scene and manually drag-and-drop each button's Frontplate's RawImage to make it work.  IMHO that's a waste of time and human-error prone.
Lastly, since I agreed to follow the framework's philosophy then I'll make the change but I think we should keep an open-mind to different approaches that are more useful to our customers in the terms of development speed and reducing human-errors.

Reply:
Ok, so, I stand corrected.  I talked to Adam offline and I now have a better understanding of this feedback.  His suggestion is to modify the Component property value (in this case the RawImage's Enabled property) directly in the Inspector's event call.  I, wrongfully, was focused on having to do it via a script but now that I better understand the feedback, I'll do it that way.

Reply:
So ... in last commit I have removed the ExperimentalPressableButtonWithDynamicFrontPlate script and also made the Frontplate's RawImage to be {enabled|disabled} on ProximityHovered{Entered|Exited} events respectively.  This change make the FrontPlateRawImage serializable field no longer necessary.  Resolving this.